axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/axon/DefaultPersistentStreamMessageSourceFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.axonserver.connector.event.axon;
+
+import io.axoniq.axonserver.connector.event.PersistentStreamProperties;
+import org.axonframework.config.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.ScheduledExecutorService;
+
+public class DefaultPersistentStreamMessageSourceFactory implements PersistentStreamMessageSourceFactory {
```

Comment:
Just checking, but you introduced this as a replacement of the lambda version constructed in the `AxonServerAutoConfiguration`, right? Or do you foresee other usages?

Reply:
Hmm... it's the only usage. So it may reside event inside `AxonServerAutoConfiguration`.
But if someone wants to create the PersistentStreamMessageSource by themselves? We should push our users towards the factory, so I left it public in the axonserver connector.

What do you think?

Reply:
Well...building a `PersistentStreamMessageSource` isn't that difficult :-)
Then again, the check that a given `name` isn't used multiple times is worthwhile to share about.

As stated, I was just checking here. I don't see anything immediately wrong. Just trying to understand your reasoning to add this is all.